scipy/special/_gufuncs_docs.cpp

```diff
@@ -0,0 +1,150 @@
+const char *lpn_doc = R"(
+    Legendre function of the first kind.
+
+    Compute sequence of Legendre functions of the first kind (polynomials),
+    Pn(z) and derivatives for all degrees from 0 to n (inclusive).
+
+    See also special.legendre for polynomial class.
+
+    References
+    ----------
+    .. [1] Zhang, Shanjie and Jin, Jianming. "Computation of Special
+           Functions", John Wiley and Sons, 1996.
+           https://people.sc.fsu.edu/~jburkardt/f77_src/special_functions/special_functions.html
+
+    )";
+
+const char *lpmn_doc = R"(
+    Sequence of associated Legendre functions of the first kind.
+
+    Computes the associated Legendre function of the first kind of order m and
+    degree n, ``Pmn(z)`` = :math:`P_n^m(z)`, and its derivative, ``Pmn'(z)``.
+    Returns two arrays of size ``(m+1, n+1)`` containing ``Pmn(z)`` and
+    ``Pmn'(z)`` for all orders from ``0..m`` and degrees from ``0..n``.
+
+    This function takes a real argument ``z``. For complex arguments ``z``
+    use clpmn instead.
+
+    Parameters
+    ----------
+    z : array_like
+        Input value.
```

Comment:
You seem to have accidentally chopped out some of the parameters here and in some of the other functions. Also, for new docstrings, if something only takes floats it's good to say "array_like of float" and if it can take real or complex values, say "array_like of float or complex"

https://github.com/scipy/scipy/blob/4edfcaa3ce8a387450b6efce968572def71be089/scipy/special/_basic.py#L1720-L1729

Reply:
Actually I didn't! The gufuncs don't take `m` and `n` as parameters, only their Python counterparts do.

Reply:
right :facepalm:, please disregard

Reply:
Oh, but it looks like you didn't touch the docstrings of the Python functions, so make sure you update those.

Reply:
I'm going to leave the docstrings alone as I've left the restrictions on these functions to scalars. That way, we can introduce all the appropriate tests in a follow-up PR and remove those restrictions then.